wiki/Building-From-Source.md

```diff
@@ -65,14 +80,41 @@ If you want to edit and rebuild the code, you can skip the `prepare-deps` comman
 For the client you need install:
 
 * [Android Studio](https://developer.android.com/studio) or the [sdkmanager](https://developer.android.com/studio/command-line/sdkmanager)
-* Platform Tools 33 (Android 13)
-* Latest NDK (currently v25.1.8937393)
-* Set the environment variable `JAVA_HOME`
-  * For example on Windows: `C:\Program Files\Android\Android Studio\jre`
-* Set the environment variable `ANDROID_SDK_ROOT`
-  * For example on Windows: `%LOCALAPPDATA%\Android\Sdk`
-* Set the environment variable `ANDROID_NDK_HOME`
-  * For example on Windows: `%LOCALAPPDATA%\Android\Sdk\ndk\25.1.8937393`
+* Latest Android SDK Platform-Tools 33 (Android 13)
+* Latest Android NDK (currently v25.1.8937393)
+
+On Linux, the three mentioned developer applications can be installed on Linux from upstream.
+
+* **Arch**
+
+  ```bash
+  sudo pacman -S tandroid-sdk-platform-tools sdkmanager android-ndk
+  ```
+
+* **Gentoo**
+
+  * `dev-util/android-studio`
+  * `dev-util/android-sdk-update-manager`
+  * `dev-util/android-ndk >= 25.1`
+
+* **Debian 12 / Ubuntu 22.10 / Pop!\_OS 22.10** (Debian requires to have the `non-free` repository enabled)
+  ```bash
+  sudo apt install android-sdk-platform-tools-common sdkmanager google-android-ndk-r25b-installer
+  ```
+
+Set the environment variables:
+  * `JAVA_HOME`:
+    * Windows: `C:\Program Files\Android\Android Studio\jre`
+    * Linux: `/usr/lib/jvm/default-java/bin`
+  * `ANDROID_SDK_ROOT`:
+    * Windows: `%LOCALAPPDATA%\Android\Sdk`
+    * Linux:
+      * Debian / Ubuntu / Pop!\_OS : `~/AndroidSDK`
+      * Arch: `~/Android/Sdk`
+      * Gentoo: `~/Android`
+  * `ANDROID_NDK_HOME`:
+    * Windows: `%LOCALAPPDATA%\Android\Sdk\ndk\25.1.8937393`
+    * Linux: `/usr/lib/android-sdk/ndk`
```

Comment:
Again, I would leave out Linux examples since they vary depending on the installation method.

Reply:
What about mention of the directory differing from install to install? With moving linux mentions to their own section, such as:

> For Windows, set the environment variables:
> 
>   * `JAVA_HOME`:
>     * `C:\Program Files\Android\Android Studio\jre`
>   * `ANDROID_SDK_ROOT`:
>     * `%LOCALAPPDATA%\Android\Sdk`
>   * `ANDROID_NDK_HOME`:
>     * `%LOCALAPPDATA%\Android\Sdk\ndk\25.1.8937393`
> 
> For Linux, the directories for the environment variables can greatly differ from the type of install to where it's installed, but generally they would be as:
> 
>   * `JAVA_HOME`:
>     * `/usr/lib/jvm/default-java/bin`
>   * `ANDROID_SDK_ROOT`:
>     * Debian / Ubuntu / Pop!\_OS: `~/AndroidSDK`
>     * Arch: `~/Android/Sdk`
>     * Gentoo: `~/Android`
>   * `ANDROID_NDK_HOME`:
>     * `/usr/lib/android-sdk/ndk`

Or a link to each distro wiki's page for is better?

Reply:
@m00nwtchr Opinions?

Reply:
When installing from the packages, Arch puts android sdk/ndk stuff under `/opt/`

Reply:
I'd keep both instructions here (for the stuff the distro wiki pages miss) and link to the distro pages.

Reply:
> When installing from the packages, Arch puts android sdk/ndk stuff under `/opt/` btw

So it would be as `/opt/android-sdk`?

Reply:
Including the above requested change, how is this?

> On Linux, the specific package names for the android tools can differ from distro to distro, see up on the wiki for more information:
> * Gentoo:
>   * https://wiki.gentoo.org/wiki/Android
> * Arch:
>   * https://wiki.archlinux.org/title/Android
> * Debian:
>   * https://wiki.debian.org/AndroidStudio
> * Ubuntu:
>   * https://help.ubuntu.com/community/AndroidSDK
> * Pop!\_OS:
>   * N/A
> 
> The three mentioned developer applications can be installed from upstream; although the packages and setup responsible for the required tools can differ between distros, being:
> 
> * **Arch**
>   * Packages can vary, read up on the Arch Wiki's [Android](https://wiki.archlinux.org/title/Android) page.
> 
> * **Gentoo**
>   * `dev-util/android-studio`
>   * `dev-util/android-sdk-update-manager`
>   * `dev-util/android-ndk >= 25.1`
> 
> For Debian, it requires to have the `non-free` repository to be enabled:
> * **Debian 12 / Ubuntu 22.10 / Pop!\_OS 22.10**
>   ```bash
>   sudo apt install android-sdk-platform-tools-common sdkmanager google-android-ndk-r25b-installer
>   ```
> 
> For Windows, set the environment variables:
> * `JAVA_HOME`:
>   * Example: `C:\Program Files\Android\Android Studio\jre`
> * `ANDROID_SDK_ROOT`:
>   * Example: `%LOCALAPPDATA%\Android\Sdk`
> * `ANDROID_NDK_HOME`:
>   * Example: `%LOCALAPPDATA%\Android\Sdk\ndk\25.1.8937393`
> 
> For Linux, the correct directories for the environment variables can greatly differ depending on the type of install. See the wiki page of your distro for more information:
> * Gentoo:
>   * https://wiki.gentoo.org/wiki/Android
> * Ubuntu:
>   * https://help.ubuntu.com/community/AndroidSDK#Post-Installation_Configuration
> 
> Distro wikis that weren't listed above does not mention of environment variables, although generally they would be as:
> * `JAVA_HOME`:
>   * `/usr/lib/jvm/default-java/bin`
> * `ANDROID_SDK_ROOT`:
>   * Arch: `~/Android/Sdk`
>   * Gentoo: `~/Android`
>   * Debian / Ubuntu / Pop!\_OS: `~/AndroidSDK`
> * `ANDROID_NDK_HOME`:
>   * Arch: `/opt/android-sdk/ndk`
>   * Linux: `/usr/lib/android-sdk/ndk`

If needed, I can add a h2 tag for each section, such as "Installing Packages" and "Environment Variables"?

Reply:
I have updated the building page with a few minor changes added on top of the above comment in some-username-here1/ALVR@6a34336871f0c40f24fbf77b5a9f6e7529f5b68c

How's this?
https://github.com/some-username-here1/ALVR/blob/bfs-linux-fixes/wiki/Building-From-Source.md